pkg/util/day_duration.go

```diff
@@ -1,4 +1,8 @@
-package dataset
```

Comment:
NIT: 

Instead of `util` why don't we move to its own package? You can put it in `internal/dayduration`? Additionally you can change the `DayDuration` to `Duration` because it will be referenced as `dayduration.Duration`. Another option could be `day.Duration`.

I don't like stuffing everything into a `util` or `misc` package. I try to follow https://github.com/golang/go/wiki/CodeReviewComments#package-names

Since we already have one I'd try not to add more things there.